packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -133,32 +134,27 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 	 */
 	private readonly baseLoggingClient: ApplicationInsights;
 	private readonly config: FluidAppInsightsLoggerConfig;
-	private readonly _filters: TelemetryFilter[];
-
 	public constructor(client: ApplicationInsights, config?: FluidAppInsightsLoggerConfig) {
 		this.baseLoggingClient = client;
-		this.config = config ?? {
-			filtering: {
-				mode: "exclusive",
-				filters: [],
-			},
-		};
-
-		// Deep copy filters to internal array
-		this._filters =
-			this.config.filtering.filters?.map((filter) => structuredClone(filter)) ?? [];
-		// Validate and merge redundant filters.
-		this._filters = this.validateFilters(this._filters);
-		// Sort filters by longest namespace first.
-		this._filters.sort((a, b) => {
-			const namespaceALength = "namespacePattern" in a ? a.namespacePattern.length : 0;
-			const namespaceBLength = "namespacePattern" in b ? b.namespacePattern.length : 0;
-			return namespaceBLength - namespaceALength;
-		});
-	}
+		// Deep copy config to internal array
+		this.config = config
+			? structuredClone(config)
+			: {
+					filtering: {
+						mode: "exclusive",
+						filters: [],
+					},
+			  };
 
-	public get filters(): TelemetryFilter[] {
-		return this._filters;
+		if (this.config.filtering.filters) {
+			this.validateFilters(this.config.filtering.filters);
+			// Sort filters by longest namespace first.
+			this.config.filtering.filters.sort((a, b) => {
+				const namespaceALength = "namespacePattern" in a ? a.namespacePattern.length : 0;
+				const namespaceBLength = "namespacePattern" in b ? b.namespacePattern.length : 0;
```

Comment:
It achieves the same thing but I think it's a more common pattern in our codebase. It also has the advantage that refactoring tools will detect it. For the same reason, let's update `validateFilters()` to also do undefined checks instead of property existence checks with strings.
```suggestion
				const namespaceALength = a.namespacePattern !== undefined ? a.namespacePattern.length : 0;
				const namespaceBLength = b.namespacePattern !== undefined ? b.namespacePattern.length : 0;
```

Reply:
`a.namespacePattern !== undefined` seems to no type cast the objects but using the `"namespacePattern" in a` does. Let me know if I'm missing something there.

This is the specific error:
```
Property 'namespacePattern' does not exist on type 'TelemetryFilter'.
  Property 'namespacePattern' does not exist on type 'CategoryFilter'
```

Reply:
Ah, we have the type union... ok, yeah, for now keep it and let's create an ADO task to review this at some point (not necessarily change it, but we should have a think).

Reply:
Per our conversation I have updated to do a combination of both, `"namespacePattern" in a && a.namespacePattern !== undefined`